TWTValidation/JSON Validator/Processors/TWTJSONSchemaPrettyPrinter.m

```
@@ -0,0 +1,279 @@
+    [self setObject:[self schemaArrayFromNodeArray:node.exactlyOneOfSchemas] forKey:TWTJSONSchemaKeywordOneOf];
+    [self setObject:[self schemaFromNode:node.notSchema] forKey:TWTJSONSchemaKeywordNot];
+
+    [self setObject:node.definitions forKey:TWTJSONSchemaKeywordDefinitions];
+}
+
+
+- (id)additionalItemsOrPropertiesFromNode:(TWTJSONSchemaASTNode *)additionalNode
+{
+    if (!additionalNode) {
+        return nil;
+    }
+
+    [additionalNode acceptProcessor:self];
+
+    return [self popCurrentObject];
+}
+
+
+- (NSDictionary *)dependencyDictionaryFromNodeArray:(NSArray *)array
+{
+    if (!array) {
+        return nil;
+    }
+
+    [self pushNewObject:[[NSMutableDictionary alloc] init]];
+    for (TWTJSONSchemaDependencyASTNode *node in array) {
+        [node acceptProcessor:self];
+    }
+
+    return [self popCurrentObject];
+}
+
+
+- (NSDictionary *)schemaFromNode:(TWTJSONSchemaASTNode *)node
+{
+    if (!node) {
+        return nil;
+    }
+
+    [node acceptProcessor:self];
+    return [self popCurrentObject];
+}
+
+
+- (NSArray *)schemaArrayFromNodeArray:(NSArray *)array
+{
+    if (!array) {
+        return nil;
+    }
+
+    NSMutableArray *nodeArray = [[NSMutableArray alloc] init];
+    for (TWTJSONSchemaASTNode *node in array) {
+        [node acceptProcessor:self];
+        [nodeArray addObject:[self popCurrentObject]];
+    }
+
+    return nodeArray;
+}
+
+
+- (NSDictionary *)schemaDictionaryFromKeyValuePairNodeArray:(NSArray *)array
+{
+    if (!array) {
+        return  nil;
+    }
+
+    [self pushNewObject:[[NSMutableDictionary alloc] init]];
+    for (TWTJSONSchemaKeyValuePairASTNode *node in array) {
+        [node acceptProcessor:self];
+    }
+
+    return [self popCurrentObject];
+}
+
+
+# pragma mark - Convenience methods
+
+- (void)setObject:(id)object forKey:(NSString *)key
+{
+    if (object && [self.currentObject isKindOfClass:[NSDictionary class]]) {
```

It seems like you shouldn’t fail silently if currentObject is not a dictionary. A programmer shouldn’t be invoking this method on non-dictionaries. Just remove the self.currentObject check and let the app crash.